hpat/hiframes/api.py

```diff
@@ -533,6 +534,31 @@ def isna_overload(arr, i):
     return lambda arr, i: False
 
 
+def get_nan_mask(arr):
+    return np.zeros(len(arr), np.bool_)
+
+
+@overload(get_nan_mask)
+def get_nan_mask_overload(arr):
+
+    def get_nan_mask_via_isna_impl(arr):
+        return np.array([isna(arr, i) for i in np.arange(len(arr))])
+
+    if isinstance(arr, types.Array):
+        dtype = arr.dtype
+        if isinstance(dtype, types.Float):
+            return lambda arr: np.isnan(arr)
```

Comment:
I wouls propose to avoid lambda here and make it in the same way as `get_nan_mask_via_isna_impl`